crest/Assets/Crest/Crest/Scripts/Underwater/UnderwaterRenderer.cs

```diff
@@ -66,13 +91,16 @@ public class DebugFields
         {
             public bool _viewOceanMask = false;
             public bool _disableOceanMask = false;
+            public bool _viewStencil = false;
             public bool _disableHeightAboveWaterOptimization = false;
             public bool _disableArtifactCorrection = false;
         }
 
         Camera _camera;
         bool _firstRender = true;
 
+        internal bool IsStencilBufferRequired => _mode == Mode.GeometryVolume;
```

Comment:
could/should a stencil buffer be used in general? i guess if cam is far above water the effect is disabled completely. if cam is close to water or partially submerged then a stencil buffer could save a bunch of processing. maybe we dont care about this case enough to warrant the complexity/memory though.

Reply:
I think so. I am looking at replacing the mask with a stencil buffer completely but will wait for 2021.2. 2021.2 has some new APIs which make it a little easier to use.

Reply:
sure sounds good

Reply:
Actually yes... I should be using the stencil buffer for the mask for all volume modes except full screen (which I am not). I was thinking of using the stencil for the full screen effect itself in my reply. Will expand its usage.